src/JuMP.jl

```diff
@@ -329,6 +329,25 @@ function direct_model(backend::MOI.ModelLike)
     )
 end
 
+"""
+    direct_model(optimizer::MOI.OptimizerWithAttributes)
+
+Return a new JuMP model that instantiates a [`OptimizerWithAttributes`](@ref) to store the
+model and solve it.
+"""
+function direct_model(optimizer::MOI.OptimizerWithAttributes)
```

Comment:
Why not just 
```julia
function direct_model(factory::MOI.OptimizerWithAttributes)
    optimizer = MOI.instantiate(factory)
    return direct_model(optimizer)
end
```
This also needs a test.

Reply:
I thought it would be unnecessary to call `MOI.instantiate` since it for direct mode bridges and the incremental interface isn't compatible. I can make the changes as suggested also.

Reply:
https://github.com/jump-dev/MathOptInterface.jl/blob/30506e3757e9d2d6c21f3cdafc207c9a2ae947b9/src/instantiate.jl#L117-L125

No bridges are added if you omit the `with_bridge_type` argument.

Reply:
fixed in 764bed3, test added in 0110cac